### ISSUE TITLE
inability to add events on 29th February

### ISSUE THREAD

**Issue by geier** [pimutils/khal#353]
Using the latest git checkout, through khal-git in the AUR, I get a crash when trying to browse to the 29th February in ikhal. When trying to add an event directly using "khal new 29.02. ..." I get a "Cannot parse: '29.02. ...' error.

Originally reported by @josswright in #351

**Comment by geier** [comment]
See http://bugs.python.org/issue14157

This has been fixed in time.strptime but not in datetime.strptime.

This should be easily fixable by using time.strptime instead of datetime.strptime. I'll take care of this ASAP.

**Comment by geier** [comment]
```python
In [1]: from datetime import datetime

In [2]: import time

In [3]: datetime.strptime('29.02', '%d.%m')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-eccd185872b4> in <module>()
----> 1 datetime.strptime('29.02', '%d.%m')

ValueError: day is out of range for month

In [4]: time.strptime('29.02', '%d.%m')
Out[4]: time.struct_time(tm_year=1900, tm_mon=2, tm_mday=29, tm_hour=0, tm_min=0, tm_sec=0, tm_wday=0, tm_yday=60, tm_isdst=-1)

```

**Comment by hobarrera** [comment]
This can be "fixed" by something like:

```
t = time.strptime(string, format)
tdata = list([*t])
tdata[0] = date.today().year
return datetime(*t[0:6])
```

Not sure how [in]efficient that is.

**Comment by geier** [comment]
This is more or less what we are doing anyway, see https://github.com/geier/khal/blob/master/khal/aux.py#L165 We probably just need to replace every occurrence of datetime.strptime by time.strptime and wrap it in datetime().

**Comment by geier** [comment]
fixed with 3804fafff6e003de94a458a2d6945d55b1727810
